app/models/miq_schedule.rb

```diff
@@ -331,16 +331,22 @@ def validate_file_depot  # TODO: Do we need this if the validations are on the F
   end
 
   def verify_file_depot(params)  # TODO: This logic belongs in the UI, not sure where
-    depot_class      = FileDepot.supported_protocols[params[:uri_prefix]]
-    depot            = file_depot.class.name == depot_class ? file_depot : build_file_depot(:type => depot_class)
-    depot.name       = params[:name]
-    depot.uri        = params[:uri]
-    depot.aws_region = params[:aws_region]
+    depot_class                = FileDepot.supported_protocols[params[:uri_prefix]]
+    depot                      = file_depot.class.name == depot_class ? file_depot : build_file_depot(:type => depot_class)
+    depot.name                 = params[:name]
+    uri                        = params[:uri]
+    api_port                   = params[:swift_api_port]
+    depot.aws_region           = params[:aws_region]
+    depot.openstack_region     = params[:openstack_region]
+    depot.keystone_api_version = params[:keystone_api_version]
+    depot.v3_domain_ident      = params[:v3_domain_ident]
+    depot.security_protocol    = params[:security_protocol]
+    depot.uri                  = api_port.blank? ? uri : depot.merged_uri(uri, api_port)
     if params[:save]
       file_depot.save!
       file_depot.update_authentication(:default => {:userid => params[:username], :password => params[:password]}) if (params[:username] || params[:password]) && depot.class.requires_credentials?
-    else
-      depot.verify_credentials(nil, params.slice(:username, :password)) if depot.class.requires_credentials?
+    elsif depot.class.requires_credentials?
+      depot.verify_credentials(nil, params)
```

Comment:
Do we not need the `.slice(:username, :password)` anymore?

Reply:
No.  That actually causes a problem.  We need more params passed on than just user/passwd.